### PR TITLE
Major Changes - Cylc Install - Rose Stem

### PR DESCRIPTION
**Pull request by wxtim**
Partial response to https://github.com/cylc/cylc-doc/issues/170: Twinned with:https://github.com/cylc/cylc-rose/pull/68

- Added a Cylc install major changes page in the Cylc 8 migration guide.
- Added a link from the summary page to the major-changes/platforms page.

### CODE REVIEW

File: src/7-to-8/major-changes/cylc-install.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+plugin.
+
+To find out if you have Cylc-Rose installed:
+
+.. code-block:: bash
+
+   cylc install --help
+
+If the output contains the options ``-D``, ``-S`` and ``-O`` you probably have
+Cylc-Rose installed.
```
Comment (kinow) [inline src/7-to-8/major-changes/cylc-install.rst:101]: Don't we have another way? Perhaps some command that lists plug-ins, that we can just `grep` for rose?

**Review by kinow — APPROVED** [review]
One comment, but not a blocker. 👍 thanks for the great work on the documentation @wxtim !

**Comment by kinow** [comment]
Oh, almost forgot to mention the CI issue. @wxtim we probably need to confirm whether it's something in this change or not, before @datamel 's final review 👍 

```bash

Warning, treated as error:
/home/runner/work/cylc-doc/cylc-doc/src/7-to-8/major-changes/cylc-install.rst:110:undefined label: rose-stem
make: *** [html] Error 2
Makefile:36: recipe for target 'html' failed
Error: Process completed with exit code 2.
```

**Comment by wxtim** [comment]
> Oh, almost forgot to mention the CI issue. @wxtim we probably need to confirm whether it's something in this change or not, before @datamel 's final review +1
> 
> ```shell
> Warning, treated as error:
> /home/runner/work/cylc-doc/cylc-doc/src/7-to-8/major-changes/cylc-install.rst:110:undefined label: rose-stem
> make: *** [html] Error 2
> Makefile:36: recipe for target 'html' failed
> Error: Process completed with exit code 2.
> ```

Caused by these changes without https://github.com/cylc/cylc-rose/pull/68 being merged to master.

File: src/7-to-8/major-changes/cylc-install.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+You can also use ``-C`` (or ``--directory``) to set a source path:
+
+.. code-block:: bash
+
+   cylc install -C /path/to/another-workflow
+
+Once you have installed a workflow you can use Cylc play to set it in motion.
+
+.. TODO - Include links to Cylc Play documentation.
```
Comment (datamel) [inline src/7-to-8/major-changes/cylc-install.rst:75]: Did you want to do these links with this PR?

File: src/7-to-8/major-changes/cylc-install.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Cylc Rose
+---------
+
+.. seealso::
+
+   :ref:`cylc-rose` plugin documentation, which includes examples of how
+   to install a Cylc workflow with a Rose suite configuration with ``cylc install``.
+
+   .. TODO link to rose stem doc once it's done
```
Comment (datamel) [inline src/7-to-8/major-changes/cylc-install.rst:86]: ditto

File: src/7-to-8/major-changes/cylc-install.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+.. code-block:: bash
+
+   > cylc install my-workflow
+   INSTALLED my-workflow/run2 from /home/me/cylc-src/my-wf
+
+.. note::
+
+   Each time you run ``cylc install`` a new copy of the workflow is installed
+   in a new directory:
```
Comment (datamel) [inline src/7-to-8/major-changes/cylc-install.rst:63]:
```suggestion
   in a new directory,
```

File: src/7-to-8/major-changes/cylc-install.rst [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+--------
+
+Cylc 7 ran workflows in ``~/cylc-run/``. You could develop your
+workflow in ``~/cylc-run`` or copy it after developing it elsewhere.
+If you developed in the ``~/cylc-run`` directory there was a risk that
+Cylc might alter your files. If you developed elsewhere you needed to
+install your workflow with another tool.
+
+We designed Cylc 8 to help you keep your development and
+running copies separated. By default you can now develop workflows in the
```
Comment (datamel) [inline src/7-to-8/major-changes/cylc-install.rst:32]:
```suggestion
running copies separate. By default you can now develop workflows in the
```

**Review by datamel — APPROVED** [review]
Minor suggestions, no blockers.
Thanks @wxtim.

**Pull request merged**